### PR TITLE
List pack dependents

### PR DESCRIPTION
**Pull request by perryqh**
Answers the question: what packs depend on the provided pack?

`pks list-pack-dependents <pack_name>` will list the "public" dependents (better name?) and the dependents in violation (not using public API).

### CODE REVIEW

File: src/packs/dependents.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+use super::Configuration;
+
+type PackName = String;
+type ViolationType = String;
+type ViolationCount = usize;
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct Dependents {
+    pub public_dependents: Vec<PackName>,
```
Comment (alexevanczuk) [inline src/packs/dependents.rs:11]: I think we might be conflating privacy and dependency checking here. This isn't about whether something is public or not, but whether the other pack has explicitly listed the pack they use as a dependency. If they haven't listed it, it's still a dependency, but an implicit one (in violation).

What do you think of calling these `explicit` and `implicit` dependencies/dependents?

**Review by alexevanczuk — COMMENTED** [review]
Hey looks good, but have one suggestion on naming, let me know what you think.

**Comment by alexevanczuk** [comment]
Thanks for your patience on the review for this one 🙏🏼 feel free to ping me if I haven't gotten to something in a bit

File: src/packs/dependents.rs (hunk 1 above)
Comment (perryqh) [reply src/packs/dependents.rs:11]: Great idea!

**Comment by perryqh** [comment]
I made the requested changes. Thanks for the careful review @alexevanczuk! 

I'm converting this to "draft" until I get back from vacation.

**Review by alexevanczuk — APPROVED** [review]
Looks great! I'm going to merge this even though it's on draft and you're OOO since it looks good, tests pass, and there is no version bump in this PR, so it won't auto-release anyways.

**Pull request merged**